packages/react-query/src/__tests__/useQuery.types.test.tsx

```diff
@@ -156,3 +156,24 @@ describe('initialData', () => {
     })
   })
 })
+
+describe('placeholderData', () => {
+  it('should receive contextual parameter type', () => {
```

Comment:
This shows how this is an improvement in the general sense, and not just for `queryOptions`. This test would fail today: [TS playground](https://www.typescriptlang.org/play?ts=5.2.2#code/JYWwDg9gTgLgBAbzgVwM4FMCKz1QJ5wC+cAZlBCHAEQACMAhgHaoMDGA1gPRTr2swBaAI458VANxxOnODDxh0qAFxwArADoADFoEAjdA3UAmVQFgAUBdYRm8CGBjAbqOAF5EFuHBG48AaXQ8FQBtKhIICCoAXTh6F2tbABpPb1E8ADFGFQAKAEo3AD44AAVyEGAMdR5UCAAbADd0bNVc5PMvFnpa9AAVUHQVAEZNOAAqOGHNTTbCcQsLNCw07IQU9XX7R2c2rzBavnQACzqAE1wAEXoGHLAeeqc0S4Z81yLV9q84W-R7iEer+hzD5eaRwAB6AH4Ul4eDBkFBGF87g9UE9ASlCDNckCgA)